src/shared/user-record.c

```diff
@@ -2027,7 +2027,7 @@ int user_record_test_blocked(UserRecord *h) {
 
         n = now(CLOCK_REALTIME);
         if (h->last_change_usec != UINT64_MAX &&
-            h->last_change_usec > n) /* Don't allow log ins when the record is from the future */
+            h->last_change_usec > n) /* Complain during log-ins when the record is from the future */
```

Comment:
... shouldn't the order of checks be changed so that -ESTALE is returned only if the other "more serious" conditions don't match?

Reply:
true!